src/llama_engine.cc

```diff
@@ -835,31 +836,53 @@ void LlamaEngine::HandleInferenceImpl(
       LOG_INFO << "Request " << request_id << ": " << "Inference completed";
     });
   } else {
+    int n = completion.n;
     auto state = CreateInferenceState(si.ctx);
-    si.q->runTaskInQueue([this, request_id, state, cb = std::move(callback),
+    si.q->runTaskInQueue([this, n, request_id, state, cb = std::move(callback),
                           d = std::move(data)]() {
       Json::Value respData;
-      int task_id = state->llama.RequestCompletion(d, false, false, -1);
+      std::vector<int> task_ids;
+      for (int i = 0; i < n; i++) {
+        task_ids.push_back(state->llama.RequestCompletion(d, false, false, -1));
+      }
       LOG_INFO << "Request " << request_id << ": "
                << "Non stream, waiting for respone";
       if (!json_value(d, "stream", false)) {
         bool has_error = false;
         std::string completion_text;
-        TaskResult result = state->llama.NextResult(task_id);
-        if (!result.error && result.stop) {
-          int prompt_tokens = result.result_json["tokens_evaluated"];
-          int predicted_tokens = result.result_json["tokens_predicted"];
-          std::string to_send = result.result_json["content"];
-          llama_utils::ltrim(to_send);
-          respData = CreateFullReturnJson(
-              llama_utils::generate_random_string(20), "_", to_send, "_",
-              prompt_tokens, predicted_tokens);
-        } else {
-          bool has_error = true;
-          respData["message"] = "Internal error during inference";
-          LOG_ERROR << "Request " << request_id << ": "
-                    << "Error during inference";
+        int prompt_tokens = 0;
+        int predicted_tokens = 0;
+        std::vector<TaskResult> results;
+        for (int i = 0; i < n; i++) {
+          results.push_back(state->llama.NextResult(task_ids[i]));
         }
+        // TaskResult result = state->llama.NextResult(task_id);
+        for (auto& result : results) {
+          if (!result.error && result.stop) {
+            prompt_tokens += result.result_json["tokens_evaluated"].get<int>();
+            predicted_tokens +=
+                result.result_json["tokens_predicted"].get<int>();
+            std::string to_send = result.result_json["content"];
+            llama_utils::ltrim(to_send);
+            if (respData.empty()) {
```

Comment:
Does this mean we do not append the first result to `choices`?

Reply:
respData is created as empty Json::Value, if respData is empty we will directly assign result to it

Reply:
Got it. Thank you,